### PR TITLE
Enable use of this library in the browser.

### PR DESCRIPTION
**Pull request by dlongley**
Still needs karma tests.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -3,13 +3,25 @@
...
  */
 'use strict';
 
-const assert = require('assert-plus');
+const env = require('./env');
+let assert;
+let URL;
+if(env.nodejs) {
+  assert = require('assert-plus');
+  URL = require('url').URL;
```
Comment (mattcollier) [inline lib/index.js]: elsewhere, I did `global.URL = require(..)` I was not sure of what the impacts of doing `let URL` were in the browser?  What are the effects?

**Comment by davidlehn** [comment]
Would be nice if we could avoid this `env.js` pattern.  It will put in a bit of the `process` polyfill for the browser and leave around dead code (I think).  I'm not sure what the best approach is.  One choice is to put node specific code in a file, `platform.js` or similar, and use the package.json `browser` field to override that to a `platform-browser.js` file.  Just export the same API from both and use platform specific code as needed.  Not always the best approach but might work here.

File: lib/index.js (hunk 1 above)
Comment (dlongley) [reply lib/index.js]: Yeah, was a mistake, it should have said `Url`. Fixed.

**Comment by dlongley** [comment]
@davidlehn, I'm happy if someone wants to do optimizations like that ... I'm just yak shaving at the moment to get things working.

**Pull request merged**